### PR TITLE
Add ActionSpan component

### PR DESCRIPTION
**Pull request by belcurv**
### What

Adds an `ActionSpan` component based on the `<span>`s in my last PR and the `<action>` tags in PR #18

### CODE REVIEW

File: src/pages/about.js [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
 
 export default function About () {
   const [isPaused, setIsPaused] = useState(true)
 
-  const togglePause = () => setIsPaused(paused => !paused)
+  const actionProps = {
+    onClick: () => setIsPaused(paused => !paused),
+    onMouseEnter: () => setIsPaused(false),
+    onMouseLeave: () => setIsPaused(true)
+  }
```
Comment (belcurv) [inline src/pages/about.js:15]: In this case, every span should trigger the same actions. And the only action is _pause_.

A more complex canvas drawing might have multiple shapes we need to animate separately. I think this system can still support that. As long as each `ActionSpan` has a unique `id`, we can update the state of specific canvas things based on their IDs

```jsx
const [actions, setActions = useState({
  pointC: { isPaused: true },
  lineCE: { isPaused: true }
})

const actionProps = {
  onClick: (event) => setIsPaused(prevState => ({
    ...prevState,
    [event.target.id]: {
      paused: !prevState[event.target.id]
    }
  }))
}
```

File: src/pages/about.js [hunk 2]
```diff
@@ -24,75 +29,24 @@ export default function About () {
...
-            <span
-              onClick={togglePause}
-              onMouseEnter={() => setIsPaused(false)}
-              onMouseLeave={() => setIsPaused(true)}
-              style={{ color: 'teal', fontWeight: 'bold', textDecoration: 'underline' }}
-            >
-              molestie
-            </span>
-            {' '}
-            nunc.  Morbi tristique senectus et netus et malesuada fames ac. Turpis massa sed elementum tempus. Non quam lacus suspendisse faucibus interdum posuere lorem ipsum dolor. Dignissim diam quis enim lobortis scelerisque fermentum. Viverra maecenas accumsan lacus vel facilisis volutpat est velit egestas. Mattis ullamcorper velit sed ullamcorper morbi tincidunt. Dapibus ultrices in iaculis nunc sed augue lacus. Sollicitudin aliquam ultrices sagittis orci a. Dui sapien eget mi proin sed libero enim. Lectus nulla at volutpat diam ut venenatis. Tellus cras adipiscing enim eu turpis egestas pretium aenean. Vitae et leo duis ut diam quam nulla porttitor massa. Tristique risus nec feugiat in. Quam elementum pulvinar etiam non. Diam maecenas sed enim ut sem. Sodales ut eu sem integer vitae justo eget magna fermentum.
```
Comment (belcurv) [inline src/pages/about.js:81]: I deleted a bunch of this noise. All this additional content was useless and obscured the sticky footer fixes from my last PR

image

File: src/components/Typography/ActionSpan.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    onMouseEnter={props.onMouseEnter}
+    onMouseLeave={props.onMouseLeave}
+  >
+    {text}
+  </span>
+)
+
+ActionSpan.propTypes = {
+  text: PropTypes.string.isRequired,
+  id: PropTypes.string,
```
Comment (belcurv) [inline src/components/Typography/ActionSpan.jsx:22]: We probably want to make `id` required:

https://github.com/peterjmartinson/huygens-site/pull/19/files#r1451918558

**Review by peterjmartinson — APPROVED** [review]
Great!

**Pull request merged**